unlock-wordpress-plugin/inc/classes/fullpostpage/class-unlock-box-fullpp.php

```diff
@@ -0,0 +1,196 @@
+			$locks[] = array(
+				'address' => $lock['lockAddress'],
+				'network' => $network_id,
+			);
+		}
+
+		// Pass the reconstructed attributes to the render_fullpp_content function
+		$new_attributes = array(
+			'locks' => $locks,
+			'ethereumNetworks' => $formattedNetworks,
+		);
+		return $this->render_fullpp_content($new_attributes, $content);
+		
+	}
+
+	/**
+	 * Render block.
+	 *
+	 * @param array  $attributes List of attributes passed in block.
+	 * @param string $content Block Content.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return string HTML elements.
+	 */
+	public function render_fullpp_content( $attributes, $content ) {
+		// Bail out if current user is admin or the author.
+		if ( current_user_can( 'manage_options' ) || ( get_the_author_meta( 'ID' ) === get_current_user_id() ) ) {
+			return $content;
+		}
+
+		if (
+			! is_user_logged_in() ||
+			( is_user_logged_in() && ! up_get_user_ethereum_address() )
+		) {
+			$login_button_text       = up_get_general_settings( 'login_button_text', __( 'Login with Unlock', 'unlock-protocol' ) );
+			$login_button_bg_color   = up_get_general_settings( 'login_button_bg_color', '#000' );
+			$login_button_text_color = up_get_general_settings( 'login_button_text_color', '#fff' );
+			$blurred_image_activated = wp_validate_boolean( up_get_general_settings( 'login_blurred_image_button', false ) );
+
+			$template_data = array(
+				'login_url'               => Unlock::get_login_url( get_permalink() ),
+				'login_button_text'       => $login_button_text,
+				'login_button_bg_color'   => $login_button_bg_color,
+				'login_button_text_color' => $login_button_text_color,
+				'blurred_image_activated' => $blurred_image_activated,
+			);
+
+			// Fetching some more data if blurred image button type is activated.
+			if ( $blurred_image_activated ) {
+				$login_button_description = up_get_general_settings( 'login_button_description', __( 'To view this content please', 'unlock-protocol' ) );
+				$login_bg_image           = up_get_general_settings( 'login_bg_image' );
+
+				$template_data['login_button_description'] = $login_button_description;
+				$template_data['login_bg_image']           = $login_bg_image;
+			}
+
+			$html_template = unlock_protocol_get_template( 'login/button', $template_data );
+
+			return apply_filters( 'unlock_protocol_login_content', $html_template, $template_data );
+		}
+
+		$locks = $attributes['locks'];
+
+		$settings = get_option( 'unlock_protocol_settings', array() );
+		$networks = isset( $settings['networks'] ) ? $settings['networks'] : array();
+
+		if ( Unlock::has_access( $networks, $locks ) ) {
+			return $content;
+		}
+
+		return $this->get_checkout_url( $attributes );
+	}
+
+	/**
+	 * Get checkout url for block
+	 *
+	 * @param array $attributes attributes.
+	 * @param array $networks networks from configuration.
+	 *
+	 * @return mixed|void
+	 */
+	private function get_checkout_url( $attributes ) {
```

Comment:
This does not reallyt get the checkout URL but render the checkout content, so let's rename to `render_checkout_content`

Reply:
We should also re-use what the plugin already does for blocks.

Reply:
Yeah, all the render_block and get_checkout_url functions codes and the naming convention are used as it was used 100% in the block, I did not add or remove a code from it.